src/include/processor/mapper/plan_mapper.h

```diff
@@ -151,6 +151,7 @@ class PlanMapper {
     storage::MemoryManager* memoryManager;
     ExpressionMapper expressionMapper;
     catalog::Catalog* catalog;
+    uint64_t numThreadsForExecution;
```

Comment:
Added this variable here because the `MorselDispatcher` needs it to initialize the active Shared State vector at compile time (fixed size)

The numThreads is not stored explicitly anywhere in the operator, since the vector needs to be defined at compile time it needs to be passed from the `PlanMapper`

Reply:
This should be removed though. MorselDispatcher is a global state and it should get numThreads through `initGloablState(context)`. The numThreads is stored in context

Reply:
Removed